static/js/topic_page/map_tile.tsx

```diff
@@ -0,0 +1,260 @@
+  metadataMap: Record<string, StatMetadata>;
+  populationData: StatApiResponse;
+}
+
+interface ChartData {
+  dataValues: { [dcid: string]: number };
+  metadata: { [dcid: string]: DataPointMetadata };
+  sources: Set<string>;
+  geoJsonData: GeoJsonData;
+  chartTitle: string;
+}
+
+export function MapTile(props: MapTilePropType): JSX.Element {
+  const [rawData, setRawData] = useState<RawData | undefined>(null);
+  const [chartData, setChartData] = useState<ChartData | undefined>(null);
+  const sourcesJsx = chartData ? getSourcesJsx(chartData.sources) : [];
+
+  useEffect(() => {
+    fetchData(
+      props.placeDcid,
+      props.enclosedPlaceType,
+      props.statVarDcid,
+      setRawData,
+      props.denomDcid
+    );
+  }, []);
+
+  useEffect(() => {
+    if (rawData) {
+      processData(
+        rawData,
+        !_.isEmpty(props.denomDcid),
+        props.chartTitle,
+        setChartData,
+        props.scaling
+      );
+    }
+  }, [rawData]);
+
+  useEffect(() => {
+    if (chartData) {
+      drawMap(chartData, props);
+    }
+  }, [chartData]);
+
+  return (
+    <div className="chart-container">
+      {chartData && (
+        <>
+          <div className="map-title">
+            <h4>{chartData.chartTitle}</h4>
+          </div>
+          <div
+            id={props.tileId}
+            className="svg-container"
+            ref={SVG_CONTAINER_ELEMENT}
+          ></div>
+          <div className="map-footer">
+            <div className="sources">Data from {sourcesJsx}</div>
+          </div>
+        </>
+      )}
+    </div>
+  );
+}
+
+function fetchData(
+  placeDcid: string,
+  enclosedPlaceType: string,
+  statVarDcid: string,
+  setRawData: (data: RawData) => void,
+  denomDcid?: string
+): void {
+  const geoJsonDataPromise = axios
+    .get(
+      `/api/choropleth/geojson?placeDcid=${placeDcid}&placeType=${enclosedPlaceType}`
+    )
+    .then((resp) => resp.data);
+  const dataDate = getCappedStatVarDate(statVarDcid);
+  const dateParam = dataDate ? `&date=${dataDate}` : "";
+  const enclosedPlaceDataPromise: Promise<GetStatSetResponse> = axios
+    .get(
+      `/api/stats/within-place?parent_place=${placeDcid}&child_type=${enclosedPlaceType}&stat_vars=${statVarDcid}${dateParam}`
+    )
+    .then((resp) => resp.data);
+  const populationPromise: Promise<StatApiResponse> = denomDcid
+    ? axios
+        .get(
+          `/api/stats/set/series/within-place?parent_place=${placeDcid}&child_type=${enclosedPlaceType}&stat_vars=${denomDcid}`
+        )
+        .then((resp) => resp.data)
+    : Promise.resolve({});
+  Promise.all([geoJsonDataPromise, enclosedPlaceDataPromise, populationPromise])
+    .then(([geoJsonData, placeStatData, populationData]) => {
+      setRawData({
+        geoJsonData,
+        placeStatData: placeStatData.data[statVarDcid],
+        metadataMap: placeStatData.metadata,
+        populationData,
+      });
+    })
+    .catch(() => {
+      setRawData(null);
```

Comment:
Add a TODO for an error message in this case?

Reply:
done